popdat/src/import_census.rs

```diff
@@ -1,11 +1,126 @@
-use map_model::Map;
-
 use crate::CensusArea;
+use abstutil::Timer;
+use geo::algorithm::intersects::Intersects;
+use geojson::GeoJson;
+use map_model::Map;
+use std::convert::TryFrom;
 
 impl CensusArea {
-    pub fn find_data_for_map(_map: &Map) -> Result<Vec<CensusArea>, String> {
-        // TODO importer/src/utils.rs has a download() helper that we could copy here. (And later
-        // dedupe, after deciding how this crate will integrate with the importer)
-        todo!()
+    pub fn find_data_for_map(map: &Map, timer: &mut Timer) -> anyhow::Result<Vec<CensusArea>> {
+        // TODO eventually it'd be nice to lazily download the info needed. For now we expect a
+        // prepared geojson file to exist in data/system/<city>/population_areas.geojson
+        //
+        // expected geojson formatted contents like:
+        // {
+        //     "type": "FeatureCollection",
+        //     "features": [
+        //          {
+        //              "type": "Feature",
+        //              "properties": { "population": 123 },
+        //              "geometry": {
+        //                  "type": "MultiPolygon",
+        //                  "coordinates": [ [ [ [ -73.7, 40.8 ], [ -73.7, 40.8 ], ...] ] ] ]
+        //               }
+        //          },
+        //          {
+        //              "type": "Feature",
+        //              "properties": { "population": 249 },
+        //              "geometry": {
+        //                  "type": "MultiPolygon",
+        //                  "coordinates": [ [ [ [ -73.8, 40.8 ], [ -73.8, 40.8 ], ...] ] ]
+        //               }
+        //           },
+        //          ...
+        //      ]
+        // }
+        //
+        // Note: intuitively you might expect a collection of Polygon's rather than  MultiPolygons,
+        // but anecdotally, the census data I've seen uses MultiPolygons. In practice almost
+        // all are MultiPoly's with just one element, but some truly have multiple polygons.
+        //
+        // When we implement downloading, importer/src/utils.rs has a download() helper that we
+        // could copy here. (And later dedupe, after deciding how this crate will integrate with
+        // the importer)
+        let path = abstutil::path(format!(
+            "system/{}/population_areas.geojson",
+            map.get_name().city
+        ));
+        let bytes = abstutil::slurp_file(&path).map_err(|s| anyhow!(s))?;
```

Comment:
Ah, I guess `String` doesn't implement `std::error::Error`, so we need the conversion here. I guess a good refactor would change all of the result types in all abst code to use `anyhow::Result`.

Reply:
yep exactly.